sql/core/src/main/scala/org/apache/spark/sql/jdbc/MsSqlServerDialect.scala

```diff
@@ -73,6 +73,36 @@ private case class MsSqlServerDialect() extends JdbcDialect with NoLegacyJDBCErr
       }
     }
 
+    override def visitCaseWhen(children: Array[String]): String = {
```

Comment:
does it really work? How do we know if this CASE WHEN returns boolean?

Reply:
good catch, fixing it

Reply:
fixed